acousticbrainz/data/__init__.py

```diff
@@ -1,192 +1,39 @@
 
-def count_lowlevel(mbid):
-    """Count number of stored low-level submissions for a specified MBID."""
-    conn = psycopg2.connect(current_app.config['PG_CONNECT'])
-    cur = conn.cursor()
-    try:
-        cur.execute("SELECT count(*) FROM lowlevel WHERE mbid = %s",
-                    (str(mbid),))
-        return cur.fetchone()[0]
 
-    except psycopg2.IntegrityError as e:
-        raise BadRequest(e)
-    except psycopg2.OperationalError as e:
-        raise ServiceUnavailable(e)
+def commit():
+    """Commits any pending transaction to the database.
 
-    return InternalServerError("Ouch!")
+    See http://initd.org/psycopg/docs/connection.html#connection.commit.
+    """
+    return _connection.commit()
 
 
-def get_summary_data(mbid):
-    """Fetches the low-level and high-level features from for the specified MBID."""
-    summary = {}
-    mbid = str(mbid)
-    conn = psycopg2.connect(current_app.config['PG_CONNECT'])
-    cur = conn.cursor()
+def init_connection(dsn):
+    global _connection
     try:
-        cur.execute("SELECT data FROM lowlevel WHERE mbid = %s", (mbid, ))
-        if not cur.rowcount:
-            raise NoDataFoundException("Can't find low-level data for this recording.")
-
-        row = cur.fetchone()
-        lowlevel = row[0]
-        if 'artist' not in lowlevel['metadata']['tags']:
-            lowlevel['metadata']['tags']['artist'] = ["[unknown]"]
-        if 'release' not in lowlevel['metadata']['tags']:
-            lowlevel['metadata']['tags']['release'] = ["[unknown]"]
-        if 'title' not in lowlevel['metadata']['tags']:
-            lowlevel['metadata']['tags']['title'] = ["[unknown]"]
-
-        lowlevel['metadata']['audio_properties']['length_formatted'] = \
-            time.strftime("%M:%S", time.gmtime(lowlevel['metadata']['audio_properties']['length']))
-
-        summary['lowlevel'] = lowlevel
-
-        cur.execute("SELECT hlj.data "
-                    "FROM highlevel hl, highlevel_json hlj "
-                    "WHERE hl.data = hlj.id "
-                    "AND hl.mbid = %s", (mbid, ))
-        if cur.rowcount:
-            summary['highlevel'] = cur.fetchone()[0]
-            try:
-                summary['genres'], summary['moods'], summary['other'] = \
-                    interpret_high_level(summary['highlevel'])
-            except KeyError:
-                pass
-
-        return summary
-
-    except psycopg2.IntegrityError as e:
-        raise BadRequest(e)
+        _connection = psycopg2.connect(dsn, cursor_factory=psycopg2.extras.DictCursor)
     except psycopg2.OperationalError as e:
-        raise ServiceUnavailable(e)
-
-    return InternalServerError("whoops!")
+        logging.error("Failed to initialize database connection: %s" % str(e))
 
 
 def run_sql_script(sql_file_path):
-    cursor = psycopg2.connect(current_app.config['PG_CONNECT']).cursor()
-    with open(sql_file_path) as sql:
-        cursor.execute(sql.read())
+    global _connection
```

Comment:
You don't need global when you just read a variable - it's only needed when assigning